apps/namadillo/src/App/Masp/ShieldedFungibleTable.tsx

```diff
@@ -1,84 +1,121 @@
+  ],
+  logo_URIs: { svg: namadaShieldedSvg },
+};
+
+const traceToDenom = (trace: string): string =>
+  trace.split("/").at(-1) ?? trace;
+
+export const ShieldedFungibleTable = ({
+  data,
+}: {
+  data: Balance;
+}): JSX.Element => {
   const [page, setPage] = useState(initialPage);
+  const { data: chainTokens } = useAtomValue(chainTokensAtom);
+  const knownChains = useAtomValue(knownChainsAtom);
 
-  // TODO
-  const query = mockData();
+  const list = useMemo(() => {
+    const addressToDenom: Record<string, string> = {};
+    chainTokens?.forEach((token) => {
+      addressToDenom[token.address] =
+        "trace" in token ? traceToDenom(token.trace) : "nam";
+    });
 
-  const list = query.data ?? [];
+    const denomToAsset: Record<string, Asset> = {
+      // TODO namadaAsset should be returned from knownChains
+      nam: namadaAsset,
+    };
+    knownChains.forEach((chain) => {
+      chain.assets.assets.forEach((asset) => {
+        asset.denom_units.forEach((unit) => {
+          denomToAsset[unit.denom] = asset;
+        });
+      });
+    });
+
+    return data.map(([address, amount]) => {
+      const denom = addressToDenom[address];
+      const asset = denomToAsset[denom];
+      return {
+        name: asset?.symbol ?? "?",
+        icon: asset?.logo_URIs?.svg ?? asset?.logo_URIs?.png,
+        balance: new BigNumber(amount),
+        dollar: new BigNumber(0), // TODO
+        ssrRate: new BigNumber(0), // TODO
+      };
+    });
+  }, [data, chainTokens, knownChains]);
 
   const headers = [
     "Token",
-    {
-      children: "Balance",
-      className: "text-right",
-    },
-    {
-      children: "SSR Rate",
-      className: "text-right",
-    },
+    { children: "Balance", className: "text-right" },
+    { children: "SSR Rate", className: "text-right" },
   ];
 
-  const renderRow = (token: MockedToken): TableRow => {
+  const renderRow = (token: TokenRow): TableRow => {
     return {
       cells: [
         <div key={`token-${token.name}`} className="flex items-center gap-4">
-          <div
-            className={"aspect-square w-8 rounded-full border border-yellow"}
-          />
+          <div className="aspect-square w-8 h-8">
+            {token.icon ?
+              <img src={token.icon} />
+            : <div className="rounded-full h-full border border-white" />}
+          </div>
           {token.name}
         </div>,
         <div
           key={`balance-${token.name}`}
           className="flex flex-col text-right leading-tight"
         >
-          <NamCurrency amount={9999.99} />
-          <span className="text-neutral-600 text-sm">$9999</span>
+          <Currency
+            amount={token.balance}
+            currency={{ symbol: token.name }}
+            currencyPosition="right"
+            spaceAroundSymbol={true}
+            hideBalances={false}
+          />
+          <span className="text-neutral-600 text-sm">
+            {formatCurrency("USD", token.dollar)}
```

Comment:
I think in the past we had something like `<FiatCurrency />` available. In my opinion, it would be great to wrap these currency displays in components due to accessibility helpers we have